addons/web_editor/tools.py

```diff
@@ -144,5 +149,36 @@ def get_video_thumbnail(video_url):
         response = requests.get(f'https://www.instagram.com/p/{video_id}/media/?size=t', timeout=10)
 
     if response and response.ok:
-        return image_process(response.content)
+        return base64.b64encode(image_process(response.content, size=(210, 120) if resize else (0, 0), crop=True if resize else False))
+    return None
+
+
+def get_video_name(video_url):
+    """ Computes the valid video name from a given URL
+        (or None in case of invalid URL).
+    """
+    source = get_video_source_data(video_url)
+    if source is None:
+        return None
+
+    res = None
+    platform, video_id = source[:2]
+
+    if platform == 'youtube':
+        params = {"format": "json", "url": "https://www.youtube.com/watch?v=%s" % video_id}
+        url = "https://www.youtube.com/oembed"
+        query_string = urllib.parse.urlencode(params)
+        url = url + "?" + query_string
+        res = requests.get(url, timeout=10)
+
+    elif platform == 'dailymotion':
+        res = requests.get(f'https://api.dailymotion.com/video/{video_id}?fields=title', timeout=10)
```

Comment:
Why doing new requests when you can get both title and thumbnail in one call?
Eg for dailymotion you can retrieve it like
```
https://api.dailymotion.com/video/x86xk9o?fields=thumbnail_url,title
```

For youtube, it seems like that url you request here can also provide the thumbnail in the same call as the title, as mention in the reply from where this code snippet comes from https://stackoverflow.com/a/52664178

I'd probably just do a `get_video_metadata()` or something method, which merges both 2 methods and performs only one get query

Reply:
Hmmm that makes sense yes, and we need them at the same time every time so I'll change it to a single call.

As for the security question, FP suggested I ping them early since they're pretty booked so they get to it eventually 😄

Reply:
Well, first, as you see, they got pinged on lines that I make you change, so they got ping for no reason / too early.

Second, that's a bit "le serpend qui se mords la queue" :smile:  :
- Don't ping them early, they will get overwhelmed
- Ok, but since they are overwhelmed, I ping early

Anyway, they got pinged so..

Also, given the task pad, I guess this task is a bit weird since it will actually be final reviewed by the website team, the other way around from how we do usually, right? Just to be sure..
It seems like this is getting reviewed by FP and AL, while specs are debated with PDE/internal team/you/AL in discord, right?

Reply:
@kzhekov oembed is a protocol that most platform try to use, so why use it only for youtube 

https://www.dailymotion.com/services/oembed
https://developer.vimeo.com/api/oembed
https://developers.facebook.com/docs/graph-api/reference/instagram-oembed/
...

Could be nice to check...

Reply:
Merged the two methods, also improved some of the API calls, we can now get Vimeo titles and instagram metadata as well as better quality thumbnails.

Reply:
Probably better to return a dict instead of a name/thumbnail tuple
Easier to later add a new return value alongside without the need of a signature change

Reply:
> Well, first, as you see, they got pinged on lines that I make you change, so they got ping for no reason / too early.
> 
> Second, that's a bit "le serpend qui se mords la queue" smile :
> 
> * Don't ping them early, they will get overwhelmed
> * Ok, but since they are overwhelmed, I ping early
> 
> Anyway, they got pinged so..
> 
> Also, given the task pad, I guess this task is a bit weird since it will actually be final reviewed by the website team, the other way around from how we do usually, right? Just to be sure.. It seems like this is getting reviewed by FP and AL, while specs are debated with PDE/internal team/you/AL in discord, right?

Oh didn't see this comment, hm weird. Yes, basically it's been a back and forth with testing it / improving the UX / working on the NGINX part which has mostly nothing to do with the actual video part but is a general improvement to serving files. I have no idea what the chain of review should be. 
The NGINX part I've already discussed with @fpodoo @antonylesuisse and @odony but there was no real review there, just suggestions and fixing problems with the implementation.
The video uploading UX and performance I've discussed with @fpodoo and @pierre-pde, and the code was partially discussed with people from the website team, but not in depth.

So overall, I think what really needs to be reviewed is the code quality in the first commit (video uploading) by the website team, and there is nothing blocking the review I think, since I took care of the latest notes and suggestions on the implementation. Current functionality should be final I think.

> Probably better to return a dict instead of a name/thumbnail tuple Easier to later add a new return value alongside without the need of a signature change

I've modified it to a dictionary, also changed the metadata endpoint to oembed for all providers, as suggested by @JKE-be.